### ISSUE TITLE
Jank noticed on app Start up

### ISSUE THREAD

**Issue by adeolaex** [flutter/flutter#60365]
Whenever a phone is restarted and an app is launched it skips some frames at certain points in the app.
An example is a when the user taps on a Button for the first time. The splash skips one or two frames.
Another example is showing a ModalBottomSheet.

All this happens only once then thereafter it's all smooth.

.Steps to reproduce

1. Run flutter create bug
2. Paste the code below in the main.dart
3. Run flutter run --release
4. Restart your device
5. Launch the app by clicking on it.

✓] Flutter (Channel master, 1.20.0-1.0.pre.209, on Mac OS X 10.15.5
    19F101, locale en-GB)
    • Flutter version 1.20.0-1.0.pre.209 at
      /Users/adeola/Downloads/flutter
    • Framework revision 5d2c8694bc (24 hours ago), 2020-06-25 12:18:02
      -0400
    • Engine revision 73ff2bdd36
    • Dart version 2.9.0 (build 2.9.0-18.0.dev 2ec47ad273)

 
Flutter doctor -v
[✓] Android toolchain - develop for Android devices (Android SDK version
    30.0.0)
    • Android SDK at /Users/adeola/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Xcode 11.6, Build version 11N700h
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • iPhone • 7301a50cae9aaf2e903ef313e632bfed64bb3704 • ios • iOS 13.6

Here is a minimal example.

`````dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MyApp(),
  );
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: Body());
  }
}

class Body extends StatefulWidget {
  Body({Key key}) : super(key: key);

  @override
  _BodyState createState() => _BodyState();
}

class _BodyState extends State<Body> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: FlatButton(
          child: Text('press me'),
          onPressed: () {
            showModalBottomSheet(
              context: context,
              builder: (context) {
                return Container(
                  child: ListView(
                    //  shrinkWrap: true,
                    children: <Widget>[
                      Theme(
                        data: ThemeData(
                          highlightColor: Colors.transparent,
                          splashColor: Colors.transparent,
                        ),
                        child: ListTile(
                          dense: true,
                          title: Column(
                            children: [
                              SizedBox(height: 4),
                              Text(
                                'E-mails are the basic credentials used.',
                                style: TextStyle(
                                    fontSize: 13.5,
                                    color: Theme.of(context)
                                        .textTheme
                                        .bodyText1
                                        .color),
                              ),
                            ],
                          ),
                          leading: Container(
                            height: double.infinity,
                            width: 60,
                            child: Icon(Icons.ac_unit),
                          ),
                          onTap: () => null,
                        ),
                      ),
                    ],
                  ),
                );
              },
            );
          },
        ),
      ),
    );
  }
}

`````

**Comment by adeolaex** [comment]
Hey @filiph . Here is the Bottom Sheet issue from the Flutter Live.

**Comment by liyuqian** [comment]
If an app has janky animations during the first run, and later becomes smooth for the same animation, then it's very likely a shader compilation jank.

More technically, a shader is a piece of program to be run on GPUs. When a shader is first used, it needs to be compiled. The compilation could cost up to a few hundred milliseconds where as a smooth frame needs to be drawn within 16 milliseconds for 60 FPS (frame-per-second) display. Hence a compilation could cause tens of frames to be missed, and drop the FPS from 60 to 6. That is the compilation jank.

One solution to mitigate is almost there https://github.com/flutter/flutter/issues/53607. I just need to write some documentation about how use the tools. I'll let you know once it's done, and see if it fixes your problem.

**Comment by adeolaex** [comment]
I see. I had a similar issue while using animations from https://rive.app in a flutter app. There was always a delay before the animations could start playing.

Just like you said  "When a shader is first used, it needs to be compiled", the flare_flutter package has a sort of warmUp function that is meant to be called inside the main() just before calling runApp() so the .flr files could be loading up before running the app thereby eradicating the delay.

I'm guessing that's what  #53607 aims to accomplish. Thank you very much for your response :)

**Comment by rostopira** [comment]
I were trying to implement simple welcome screens using PageView with 720p height pics as background + two lines of text and a button. User will see them only on first, and they work just terribly even on iPhone X. Yeah, I can try a trick with SkSL on Android, but what could I possibly do with iOS?

**Comment by adeolaex** [comment]
> I were trying to implement simple welcome screens using PageView with 720p height pics as background + two lines of text and a button. User will see them only on first, and they work just terribly even on iPhone X. Yeah, I can try a trick with SkSL on Android, but what could I possibly do with iOS?

This would greatly help https://github.com/flutter/flutter/wiki/Reduce-shader-compilation-jank-using-SkSL-warm-up

**Comment by rostopira** [comment]
@adeolaex 

> SkSL warm-up doesn't help newer iPhones with Metal.

**Comment by adeolaex** [comment]
@rostopira

 Thats sadly true. #53609 plans to make better solution so it works well with Metal.

May I ask for the command used when running `flutter build` for IOS, using SkSL warm up.

**Comment by Ufuk94** [comment]
Hi @liyuqian, I followed the [documentation](https://github.com/flutter/flutter/wiki/Reduce-shader-compilation-jank-using-SkSL-warm-up) and it's working. When can we expect a solution for iOS and Metal? The jank is just horrible in my app. I think I have to remove some animations for the iOS version until there is a fix.

**Comment by liyuqian** [comment]
@adeolaex @Ufuk94 : I've created https://github.com/flutter/flutter/issues/61045 to collect feedback on Metal iOS shader compilation jank severity. Please give us some numbers and stories there so we can make an informed decision on how to handle this for newer iPhones (e.g., iPhone X).

**Comment by chinmaygarde** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/61450
